src/Itil_Project.php

```diff
@@ -130,130 +135,113 @@ public static function showForProject(Project $project)
-                    Problem::class => __('Add a problem'),
-                    Ticket::class => __('Add a ticket'),
-                    default => null
-                };
-                echo '<tr class="tab_bg_2"><th colspan="2">' . $label . '</th></tr>';
-                echo '<tr class="tab_bg_2">';
-                echo '<td>';
-                echo '<input type="hidden" name="projects_id" value="' . $ID . '" />';
-                echo '<input type="hidden" name="itemtype" value="' . $itemtype . '" />';
-                $itemtype::dropdown(
-                    [
-                        'entity'      => $project->getEntityID(),
-                        'entity_sons' => $project->isRecursive(),
-                        'name'        => 'items_id',
-                        'used'        => $used,
-                    ]
-                );
-                echo '</td>';
-                echo '<td class="center">';
-                echo '<input type="submit" name="add" value="' . _sx('button', 'Add') . '" class="btn btn-primary" />';
-                echo '</td>';
-                echo '</tr>';
-                echo '</table>';
-                Html::closeForm();
-                echo '</div>';
-            }
-
-            echo '<div class="spaced">';
-            $massContainerId = 'mass' . __CLASS__ . $rand;
-            if ($canedit && $numrows) {
-                Html::openMassiveActionsForm($massContainerId);
-                $massiveactionparams = [
-                    'num_displayed' => min($_SESSION['glpilist_limit'], $numrows),
-                    'container'     => $massContainerId,
-                ];
-                Html::showMassiveActions($massiveactionparams);
-            }
+            ];
+        }
 
-            echo '<table class="tab_cadre_fixehov">';
-            echo '<tr class="noHover">';
-            echo '<th colspan="12">' . $itemtype::getTypeName($numrows) . '</th>';
-            echo '</tr>';
-            if ($numrows) {
-                $itemtype::commonListHeader(Search::HTML_OUTPUT, $massContainerId);
-                Session::initNavigateListItems(
-                    $itemtype,
-                    //TRANS : %1$s is the itemtype name,
-                    //        %2$s is the name of the item (used for headings of a list)
-                    sprintf(__('%1$s = %2$s'), Project::getTypeName(1), $project->fields['name'])
-                );
+        $it = $DB->request([
+            'FROM' => new \Glpi\DBAL\QueryUnion($queries),
+        ]);
+        $entries_by_itemtype = [];
+        $used  = [];
+        foreach ($it as $data) {
+            $used[$data['itemtype']][$data['id']]  = $data['id'];
+            $entries_by_itemtype[$data['itemtype']][] = [
+                'id'             => $data['linkid'],
+                'itemtype'       => $data['itemtype'],
+                'item_id'        => $data['id'],
+                'itemtype_label' => $data['itemtype_label'],
```

Comment:
It can probably be generated directly here, instead of setting the value in the SQL query.
```suggestion
                'itemtype_label' => $data['itemtype']::getTypeName(1),
```